src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -174,19 +176,19 @@ public static IMethodSymbol GetPropertyMethod (IPropertyReferenceOperation opera
 
 		public override TValue VisitPropertyReference (IPropertyReferenceOperation operation, LocalDataFlowState<TValue, TValueLattice> state)
 		{
-			if (operation.Instance != null) {
-				var instanceValue = Visit (operation.Instance, state);
-				HandleReceiverArgument (instanceValue, GetPropertyMethod (operation), operation);
+			if (operation.GetValueUsageInfo (operation.Property).HasFlag (ValueUsageInfo.Read)) {
```

Comment:
This looks right to me. The tricky case is something like

```C#
class C
{
    public int P { get; set; } = 0;
}
var c = new C();
c.P += 1; // both a read and a write in the same operation
```

Reply:
So Assignment operation is not the only place where values are assigned to property/field/variable?

@sbomer - I guess we have a lot more work to make the data flow work reliably. I guess our saving grace is that we're currently only tracking Type/string values for which operations like `+=` don't really make sense.

Reply:
One thing we probably need to do is go down the list of IOperations looking for any cases that aren't handled by `VisitSimpleAssignment` and `VisitFlowCapture`. I haven't encountered any missing cases in the tests we've enabled so far but there are almost certainly some.

Reply:
Filed https://github.com/dotnet/linker/issues/2507 to track this